corehq/ex-submodules/couchforms/management/commands/purge_forms_and_cases.py

```diff
@@ -68,8 +69,11 @@ def handle(self, *args, **options):
     def iterate_forms_and_collect_case_ids(self):
         print "Iterating Through %s XForms and Collecting Case Ids" % len(self.xform_ids)
         for xform in self.forms_accessor.iter_forms(self.xform_ids):
+            # Get app version by fetching app corresponding to xform build_id since xform.form
+            # does not have updated app version unless form was updated for that version
+            xform.app_version_built_with = self.get_xform_build_version(xform)
```

Comment:
the danger of adding an arbitrary property to the root object is that because of the way couch `Document`s work if someone later called `form.save()` it would persist that property to the database which seems like it would be an unintended side effect of this code. I don't think it is a blocker but in this instance I think you can define `app_version_built_with` as a variable and then just pass it through to the other calls where you need it to avoid that potential problem